### PR TITLE
chore: converts remaining files api to async iterators

### PR DESCRIPTION
**Pull request by achingbrain**

### CODE REVIEW

File: src/files-regular/get.js [hunk 1]
```diff
@@ -1,52 +1,48 @@
 'use strict'
 
-const promisify = require('promisify-es6')
+const configure = require('../lib/configure')
+const tarStreamToObjects = require('../utils/tar-stream-to-objects')
+const IsIpfs = require('is-ipfs')
```
Comment (daviddias) [inline src/files-regular/get.js]:
```suggestion
const isIpfs = require('is-ipfs')
```

Reserve capital case for constructors.

Comment (daviddias) [inline src/files-regular/get.js]:
```suggestion
const isIpfs = require('is-ipfs')
```

Reserve Capital for Constructors

File: src/files-regular/get.js [hunk 2]
```diff
@@ -1,52 +1,48 @@
...
+module.exports = configure(({ ky }) => {
+  return async function * get (path, options) {
+    options = options || {}
 
     try {
       path = cleanCID(path)
     } catch (err) {
-      if (!v.ipfsPath(path)) {
-        return callback(err)
+      if (!IsIpfs.ipfsPath(path)) {
```
Comment (daviddias) [inline src/files-regular/get.js]:
```suggestion
      if (!isIpfs.ipfsPath(path)) {
```

File: src/files-regular/ls.js [hunk 3]
```diff
@@ -1,64 +1,72 @@
 'use strict'
 
-const promisify = require('promisify-es6')
 const IsIpfs = require('is-ipfs')
```
Comment (daviddias) [inline src/files-regular/ls.js]:
```suggestion
const isIpfs = require('is-ipfs')
```

File: src/files-regular/ls.js [hunk 4]
```diff
@@ -1,64 +1,72 @@
...
+  return async function * ls (path, options) {
+    options = options || {}
 
     try {
-      args = cleanCID(args)
+      path = cleanCID(path)
     } catch (err) {
-      if (!IsIpfs.ipfsPath(args)) {
-        return callback(err)
+      if (!IsIpfs.ipfsPath(path)) {
```
Comment (daviddias) [inline src/files-regular/ls.js]:
```suggestion
      if (!isIpfs.ipfsPath(path)) {
```

File: src/files-regular/refs.js [hunk 5]
```diff
@@ -1,75 +1,63 @@
 'use strict'
 
-const IsIpfs = require('is-ipfs')
-const promisify = require('promisify-es6')
-const streamToValueWithTransformer = require('../utils/stream-to-value-with-transformer')
-const moduleConfig = require('../utils/module-config')
+const configure = require('../lib/configure')
 const cleanCID = require('../utils/clean-cid')
+const IsIpfs = require('is-ipfs')
```
Comment (daviddias) [inline src/files-regular/refs.js]:
```suggestion
const isIpfs = require('is-ipfs')
```

File: src/files-regular/get.js (hunk 1 above)
Comment (achingbrain) [reply src/files-regular/get.js]: Maybe it's my Java showing but this looks natural to me 🤣 

Isn't this the sort of thing we have a linter for?

Comment (daviddias) [reply src/files-regular/get.js]: wanna add that rule?

**Comment by achingbrain** [comment]
I'm going to merge this to unblock https://github.com/ipfs/js-ipfs/pull/2517 - have opened https://github.com/ipfs/aegir/issues/449 to discuss the capitalisation thing further.

The lines of code in question were pre-existing in the codebase, so let's not widen the scope of this PR.  

Happy to open another for general improvements if we decide it's a good idea.

**Pull request merged**